src/test/scala/com/fulcrumgenomics/umi/GroupReadsByUmiTest.scala

```diff
@@ -279,6 +278,37 @@ class GroupReadsByUmiTest extends UnitSpec with OptionValues with PrivateMethodT
     aIds.head should not equal bIds.head
   }
 
+  it should "correctly group fragment reads with the paired assigner when the two UMIs are the same" in {
+    // NB: the "a*" reads are grouped, and the "b*" reads are a second group
+    val builder = new SamBuilder(readLength=100, sort=Some(SamOrder.Coordinate))
+    builder.addFrag(name="a01", start=100, strand=Plus, attrs=Map("RX" -> "ACT-ACT"))
+    builder.addFrag(name="a02", start=100, strand=Plus, attrs=Map("RX" -> "ACT-ACT"))
+    builder.addFrag(name="a03", start=100, strand=Plus, attrs=Map("RX" -> "ACT-ACT"))
+    builder.addFrag(name="a04", start=100, strand=Plus, attrs=Map("RX" -> "ACT-ACT"))
+    builder.addFrag(name="b01", start=100, strand=Minus, attrs=Map("RX" -> "ACT-ACT"))
+    builder.addFrag(name="b02", start=100, strand=Minus, attrs=Map("RX" -> "ACT-ACT"))
+    builder.addFrag(name="b03", start=100, strand=Minus, attrs=Map("RX" -> "ACT-ACT"))
+    builder.addFrag(name="b04", start=100, strand=Minus, attrs=Map("RX" -> "ACT-ACT"))
```

Comment:
I think this test _only_ works because there are no other reads present.  When sorting these into TemplateCoordinate the `a` reads have a 5' end position of `100` but the `b`s have a 5' end position of 199.  If there were any other reads present with 5' ends anywhere between 100-199, then they would not get sorted together and your change to ReadInfo would have no effect.

Reply:
And I really don't think we want to change the TemplateCoordinate sort order.